src/components/Executions/Tables/NodeExecutionChildren.tsx

```diff
@@ -1,132 +1,50 @@
-import { WaitForData } from 'components/common';
-import {
-    RefreshConfig,
-    useDataRefresher,
-    useWorkflowExecution
-} from 'components/hooks';
-import { isEqual } from 'lodash';
-import { Execution, NodeExecutionClosure, WorkflowNodeMetadata } from 'models';
+import { Typography } from '@material-ui/core';
+import * as classnames from 'classnames';
+import { useExpandableMonospaceTextStyles } from 'components/common/ExpandableMonospaceText';
 import * as React from 'react';
-import { executionIsTerminal, executionRefreshIntervalMs } from '..';
-import { ExecutionContext } from '../ExecutionDetails/contexts';
-import { DetailedNodeExecution } from '../types';
-import { useDetailedTaskExecutions } from '../useDetailedTaskExecutions';
-import {
-    useTaskExecutions,
-    useTaskExecutionsRefresher
-} from '../useTaskExecutions';
-import { generateRowSkeleton } from './generateRowSkeleton';
-import { NoExecutionsContent } from './NoExecutionsContent';
-import { useColumnStyles } from './styles';
-import { generateColumns as generateTaskExecutionColumns } from './taskExecutionColumns';
-import { TaskExecutionRow } from './TaskExecutionRow';
-import { generateColumns as generateWorkflowExecutionColumns } from './workflowExecutionColumns';
-import { WorkflowExecutionRow } from './WorkflowExecutionRow';
+import { DetailedNodeExecutionGroup } from '../types';
+import { NodeExecutionRow } from './NodeExecutionRow';
+import { useExecutionTableStyles } from './styles';
 
 export interface NodeExecutionChildrenProps {
-    execution: DetailedNodeExecution;
+    childGroups: DetailedNodeExecutionGroup[];
 }
 
-const TaskNodeExecutionChildren: React.FC<NodeExecutionChildrenProps> = ({
-    execution: nodeExecution
+/** Renders a nested list of row items for children of a NodeExecution */
+export const NodeExecutionChildren: React.FC<NodeExecutionChildrenProps> = ({
+    childGroups
 }) => {
-    const taskExecutions = useDetailedTaskExecutions(
-        useTaskExecutions(nodeExecution.id)
-    );
-    useTaskExecutionsRefresher(nodeExecution, taskExecutions);
-
-    const columnStyles = useColumnStyles();
-    // Memoizing columns so they won't be re-generated unless the styles change
-    const { columns, Skeleton } = React.useMemo(() => {
-        const columns = generateTaskExecutionColumns(columnStyles);
-        return { columns, Skeleton: generateRowSkeleton(columns) };
-    }, [columnStyles]);
+    const showNames = childGroups.length > 1;
```

Comment:
nit: prefer this logic inline if only being used once on line 30

Reply:
That makes sense. I use this pattern so that it's clear _what_ the logic is being used for: determining if we show the names of the groups. When placed inline, it feels like more cognitive load to figure out that the names are conditionally shown _and_ to figure out what controls it.
But it's probably a matter of preference?